### ISSUE TITLE
Question: Does the repo files get modified in any way?

### ISSUE THREAD

**Issue by wighawag** [jborg/attic#369]
Zbackup has the following feature:
- Repository consists of immutable files. No existing files are ever modified

Is it the same for attic?

**Comment by ThomasWaldmann** [comment]
Usually, new data is appended and new files are added. Existing data is not modified.

What it does though is that if a chunk is deleted (and thus creates an unused "hole" in an existing segment), it will flag this segment for later compaction. Compaction means that it will transfer all used chunks from non-compact segment files to new, compact segment files and afterwards remove the non-compact files.

Chunks will be deleted when you delete archives (or let "prune" delete archives following some given policy). Also, the manifest will be regularly rewritten (new manifest added, old deleted) when adding or removing archives.

So, in short: it is very careful and does not modify existing data in files, but it still gives you the opportunity to delete/prune archives and keep storage needs low.

**Comment by ThomasWaldmann** [comment]
btw, this issue's title is way too long, please shorten it.

**Comment by wighawag** [comment]
ok thanks for the info
